### PR TITLE
Require Java 11 and update parent POM, BOM, and minimum supported Jenkins version

### PR DESCRIPTION
**Pull request by dwnusbaum**
Replaces #268 and #272.

(I do not particularly care about releasing these changes; I only want to see if `FileLogStorageTest` passes on Windows when running Java 11.)

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [ ] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

**Comment by dwnusbaum** [comment]
Indeed it looks like the behavior of setting the system property `line.separator` at runtime changes from Java 8 (new value is used) to Java 11 (new value is ignored). I will try to adapt the tests.

File: Jenkinsfile [hunk 1]
```diff
@@ -1,5 +1,4 @@
 buildPlugin(useContainerAgent: true, configurations: [
-  [ platform: "linux", jdk: "8" ],
-  [ platform: "windows", jdk: "8" ],
-  [ platform: "linux", jdk: "11" ]
+  [ platform: "linux", jdk: "17" ],
+  [ platform: "windows", jdk: "11" ]
```
Comment (jglick) [inline Jenkinsfile:3]: or perhaps more intuitively
```suggestion
  [platform: "linux", jdk: 17],
  [platform: "windows", jdk: 11]
```

**Review by jglick — APPROVED** [review]

File: src/test/java/org/jenkinsci/plugins/workflow/log/LogStorageTestBase.java [hunk 2]
```diff
@@ -330,6 +343,13 @@ private AnnotatedLargeText<?> text(String id) {
         return createStorage().stepLog(new MockNode(id), true);
     }
 
+    /**
+     * Concatenate the given lines after interspersing system-dependent line separators between them and adding a final line separator.
+     */
+    private String lines(CharSequence... lines) {
```
Comment (dwnusbaum) [inline src/test/java/org/jenkinsci/plugins/workflow/log/LogStorageTestBase.java:349]: Maybe [unlines](https://hackage.haskell.org/package/base-4.17.0.0/docs/Prelude.html#v:unlines) or something else would be a better name.

Comment (jglick) [reply src/test/java/org/jenkinsci/plugins/workflow/log/LogStorageTestBase.java:349]: `lines` seems intuitive as is.

**Review by jglick — APPROVED** [review]

**Review by jglick — APPROVED** [review]

**Comment by dwnusbaum** [comment]
@jglick Do you want a release of #274 with the old baseline before I merge this? Or does it not matter?

**Comment by jglick** [comment]
:eyes: three-week-old BOM, probably fine either way.

**Review by basil — APPROVED** [review]

**Pull request merged**